src/gs-image-tile.c

```diff
@@ -0,0 +1,230 @@
+				     "GnomeSoftware::ImageTile-css");
+
+	gs_image_set_from_pixbuf (GTK_IMAGE (tile->icon),
+				  gs_app_get_pixbuf (tile->app));
+
+	gtk_label_set_label (GTK_LABEL (tile->app_name), gs_app_get_name (app));
+	gtk_label_set_label (GTK_LABEL (tile->hover_app_name),
+			     gs_app_get_name (app));
+	gtk_label_set_label (GTK_LABEL (tile->app_summary),
+			     gs_app_get_summary (app));
+}
+
+static void
+gs_image_tile_destroy (GtkWidget *widget)
+{
+	GsImageTile *tile = GS_IMAGE_TILE (widget);
+
+	if (tile->app)
+		g_signal_handlers_disconnect_by_func (tile->app,
+						      app_state_changed, tile);
+
+	g_clear_object (&tile->app);
+
+	GTK_WIDGET_CLASS (gs_image_tile_parent_class)->destroy (widget);
+}
+
+static void
+gs_image_tile_init (GsImageTile *tile)
+{
+	gtk_widget_set_has_window (GTK_WIDGET (tile), FALSE);
+	gtk_widget_init_template (GTK_WIDGET (tile));
+	gs_star_widget_set_icon_size (GS_STAR_WIDGET (tile->stars), 12);
+}
+
+static gboolean
+gs_image_tile_crossing_notify_event (GtkWidget *widget,
+				     GdkEventCrossing *event)
+{
+	GtkWidgetClass *klass = GTK_WIDGET_CLASS (gs_image_tile_parent_class);
+	GsImageTile *tile = GS_IMAGE_TILE (widget);
+	gboolean type_enter = event->type == GDK_ENTER_NOTIFY;
+
+	gtk_revealer_set_reveal_child (GTK_REVEALER (tile->details_revealer),
+				       type_enter);
+
+	/* chain up */
+	if (type_enter)
+		return klass->enter_notify_event (widget, event);
+
+	return klass->leave_notify_event (widget, event);
+}
+
+static void
+gs_image_tile_class_init (GsImageTileClass *klass)
+{
+	GtkWidgetClass *widget_class = GTK_WIDGET_CLASS (klass);
+	GsAppTileClass *app_tile_class = GS_APP_TILE_CLASS (klass);
+
+	widget_class->destroy = gs_image_tile_destroy;
+	widget_class->enter_notify_event = gs_image_tile_crossing_notify_event;
+	widget_class->leave_notify_event = gs_image_tile_crossing_notify_event;
+
+	app_tile_class->set_app = gs_image_tile_set_app;
+	app_tile_class->get_app = gs_image_tile_get_app;
+
+	gtk_widget_class_set_template_from_resource (widget_class,
+					"/org/gnome/Software/gs-image-tile.ui");
+
+	gtk_widget_class_bind_template_child (widget_class, GsImageTile,
+					      app_name);
+	gtk_widget_class_bind_template_child (widget_class, GsImageTile,
+					      app_summary);
+	gtk_widget_class_bind_template_child (widget_class, GsImageTile, icon);
+	gtk_widget_class_bind_template_child (widget_class, GsImageTile,
+					      eventbox);
+	gtk_widget_class_bind_template_child (widget_class, GsImageTile,
+					      hover_app_name);
+	gtk_widget_class_bind_template_child (widget_class, GsImageTile, stack);
+	gtk_widget_class_bind_template_child (widget_class, GsImageTile, stars);
+	gtk_widget_class_bind_template_child (widget_class, GsImageTile,
+					      details_revealer);
+}
+
+GtkWidget *
+gs_image_tile_new (GsApp *app)
+{
+	GsImageTile *tile;
+
+	tile = g_object_new (GS_TYPE_IMAGE_TILE, NULL);
+	gs_app_tile_set_app (GS_APP_TILE (tile), app);
```

Comment:
See the comment I made in the other PR about keeping _new() functions as dumb as possible.